src/ooxml-file-cache.ts

```diff
@@ -324,11 +325,13 @@ export class OOXMLFileCache {
    */
   async readFile(cachedFilePath: string): Promise<Uint8Array> {
     try {
-      return await workspace.fs.readFile(Uri.file(cachedFilePath));
+      if (existsSync(cachedFilePath)) {
```

Comment:
We're returning an empty Uint8Array in order to handle the situation when 2 files are compared for the "Compare with Previous" command. If one of the files doesn't exist, because it was removed, a slide is deleted from a pptx for example, instead of throwing an error we return an empty Uint8Array, so that the existing file is compared to a blank file instead of throwing an error.

Reply:
Do you have more details on this situation.  If I use the current published version of the extension to "Compare with Previous" a deleted slide, it doesn't throw an error for me.

Reply:
You're right, it doesn't throw an error, but it should. In the `index.d.ts` for vscode the definition for readFile states it throws `FileSystemError.FileNotFound` if the uri doesn't exist:

```javascript
/**
 * Read the entire contents of a file.
 *
 * @param uri The uri of the file.
 * @return An array of bytes or a thenable that resolves to such.
 * @throws {@link FileSystemError.FileNotFound `FileNotFound`} when `uri` doesn't exist.
 */
readFile(uri: Uri): Uint8Array | Thenable<Uint8Array>;
```

So, it would be fine for now to skip checking if the file exists, but if we remove the check and `readFile` is refactored to match its definition the viewer will throw errors when xml parts are deleted.

Reply:
We should be handling the scenario where xml parts are being deleted/created in code already (not relying on whether or not `readFile` throws if a file doesn't exist). When an xml part is deleted and the red asterisk is showing, we're creating an empty file rather than deleting it in the cached files (so the vscode diff is diffing the old file a file with no contents)
https://github.com/yuenm18/ooxml-viewer-vscode/blob/91359040f52b7c44fda36a5d120bd35b011f9386/src/ooxml-viewer.ts#L445
and when an xml part gets created and the green asterisk is showing, we're also doing that (creating an empty compare file)
https://github.com/yuenm18/ooxml-viewer-vscode/blob/91359040f52b7c44fda36a5d120bd35b011f9386/src/ooxml-viewer.ts#L335

Reply:
That's true, I removed the `existsSync` check.